core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -79,22 +84,36 @@ private class ClientActor(driverArgs: ClientArguments, conf: SparkConf)
           driverArgs.supervise,
           command)
 
-        masterActor ! RequestSubmitDriver(driverDescription)
+        masterEndpoint.sendWithReply[SubmitDriverResponse](RequestSubmitDriver(driverDescription)).
+          onComplete {
+            case Success(v) => self.send(v)
+            case Failure(e) =>
+              println(s"Error sending messages to master ${driverArgs.master}, exiting.")
```

Comment:
`logError`? Feels wrong to print error messages to stdout. Especially since `e.printStackTrace()` will go to stderr, so you can end up with mixed output.